crates/cheatnet/src/execution/syscall_interceptor.rs

```diff
@@ -0,0 +1,206 @@
+use cairo_felt::Felt252;
+use cairo_vm::hint_processor::hint_processor_definition::HintReference;
+use cairo_vm::serde::deserialize_program::ApTracking;
+use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::vm::errors::hint_errors::HintError;
+use cairo_vm::vm::errors::vm_errors::VirtualMachineError;
+use cairo_vm::vm::runners::cairo_runner::RunResources;
+use cairo_vm::vm::vm_core::VirtualMachine;
+use std::any::Any;
+use std::collections::HashMap;
+
+pub struct ExecuteHintRequest<'a> {
+    pub vm: &'a mut VirtualMachine,
+    pub exec_scopes: &'a mut ExecutionScopes,
+    pub hint_data: &'a Box<dyn Any>,
+    pub constants: &'a HashMap<String, Felt252>,
+}
+
+pub trait ChainableHintProcessor {
```

Comment:
I am not sure if chainable is a good name, as chain suggest sequencial operations. Maybe something like HintProcessorExtension?